### ISSUE TITLE
Deprecate WooCommerce Icon Mixin

### ISSUE THREAD

**Issue by nerrad** [woothemes/woocommerce#40777]
### Describe the solution you'd like

WooCommerce ships with [custom icons](https://rawgit.com/woothemes/woocommerce-icons/master/demo.html) (shipped [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/css/_fonts.scss)) that hasn't been updated since 2013. It was also referenced in WooCommerce documentation (on how to use the icon mixins) that is largely irrelevant for the current WooCommerce UI. 

There is certainly no harm with keeping the WooCommerce mixins in place for now. However, as a reference point, I did a search on wpdirectory.net (a way to search the code of everything hosted on WP.org) for any plugin importing the WooCommerce mixins and there [was only one hit for plugins](https://wpdirectory.net/search/01HCN58T1B9NHFYPQBPCG1AMVW) and no matches for themes. 

You can view the [custom icon glyphs here](https://rawgit.com/woothemes/woocommerce-icons/master/demo.html).

Some suggested next steps:

* [x] Unpublish existing documentation referencing this mixin and the custom icons.
* [ ] Search through the Woo Core codebase to see where any custom icons might be used.
* [ ] Design review on our icon usage and what we want to expose for 3rd party use.
* [ ] Revisit how the icons are delivered for external use (ideally some something that correlates with how WP icons are available).
* [ ] Remove existing icons.

There's no urgency to this issue (why it is low priority) and I anticipate it will surface as a part of a larger admin UX update in parallel to the WordPress Phase 3 project.

**Comment by nerrad** [comment]
For reference, here's where the fonts referenced in [`_fonts.scss`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/css/_fonts.scss) actually live:

- **Shipped font files:** [`plugins/woocommerce/assets/fonts/`](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/assets/fonts) — the `../fonts/` paths in the SCSS resolve relative to the compiled CSS in `assets/css/`, not the SCSS source (there's no `fonts/` dir under `client/legacy/`).
- **Original icon font source:** [`woocommerce/woocommerce-icons`](https://github.com/woocommerce/woocommerce-icons) (the old `woothemes/woocommerce-icons` URL redirects there). It's archived and contains the IcoMoon `selection.json`, the font files, and the `demo.html` glyph reference. Untouched since ~2013.

I still think this should be picked up to:

- assess where these fonts are used.
- if they aren't used anywhere we can probably deprecate (especially since my earlier search doesn't show much usage). This would need a search across any of the extensions in the Woo Marketplace (and our owned extensions) to double-check mixins aren't in use.

**Comment by kkoteen** [comment]
Thanks for the update @nerrad! Marking as ready for engineering.
